Improve documentation

WuestFlorian opened the pull request:
Closes https://github.com/SWT-Forschungsprojekt/realtime-metrics/issues/18

WuestFlorian committed:
Improve documentation

Adapted the format of the doc strings and expanded documentation

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -13,7 +13,13 @@
+    - experienced wait time delay (defined here: https://docs.google.com/document/d/1-AOtPaEViMcY6B5uTAYj7oVkwry3LfAQJg3ihSRTVoU)
+    - availability of acceptable stop time updates (defined here: https://docs.google.com/document/d/1-AOtPaEViMcY6B5uTAYj7oVkwry3LfAQJg3ihSRTVoU)
+    - prediction reliability (defined here: https://docs.google.com/document/d/1-AOtPaEViMcY6B5uTAYj7oVkwry3LfAQJg3ihSRTVoU)
+    - prediction inconsistency (defined here: https://docs.google.com/document/d/1-AOtPaEViMcY6B5uTAYj7oVkwry3LfAQJg3ihSRTVoU)
```
Maybe it makes sense to link to the different subheadings, otherwise it is everywhere the same link

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -111,7 +117,8 @@ def run_stop_time_analysis():
     """
-    get all vehicle positions and compute the different metrics
+    Fetches all vehicle positions, computes the following metrics and prints their results to the console:
+    - availability of acceptable vehicle updates (defined here: https://docs.google.com/document/d/1-AOtPaEViMcY6B5uTAYj7oVkwry3LfAQJg3ihSRTVoU)
```
Same here, the subheading would bring someone directly to the right section

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -146,7 +153,19 @@ def run_vehicle_position_analysis():
+    Parameters
+    ----------
+    stop_time_updates : list[tuple[TripUpdate, StopTimeUpdate]
+        a list of StopTimeUpdates with the corresponding TripUpdate
```
It is already documented that it's a list, by the type hint, so no need to tell it again
```suggestion
        pairs of StopTimeUpdates and the corresponding TripUpdate
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -146,7 +153,19 @@ def run_vehicle_position_analysis():
+    Returns
+    -------
+    float | None
+        A float containing the mean squared deviation of the true arrival time. 
```
Same here, "a float containing " does not provide additional information.
```suggestion
        the mean squared deviation of the true arrival time
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -180,7 +199,24 @@ def mse_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]) ->
+    Computes the accuracy of the given stop time updates using the ETA bucketing approach.
+    The metric is defined here: https://github.com/TransitApp/ETA-Accuracy-Benchmark.
+    It sorts the stop time updates into buckets based on how much time is left until the arrival.
+    In each bucket, the allowed to be within a different range around the true arrival time.
```
Do you mean this?
```suggestion
    In each bucket, the predictions are allowed to be within a different range around the true arrival time.
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -180,7 +199,24 @@ def mse_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]) ->
+    Parameters
+    ----------
+    stop_time_updates : list[tuple[TripUpdate, StopTimeUpdate]
+        a list of StopTimeUpdates with the corresponding TripUpdate
```
```suggestion
        pairs of StopTimeUpdates with the corresponding TripUpdate
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -180,7 +199,24 @@ def mse_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]) ->
+    Returns
+    -------
+    float | None
+        A float containing the mean squared deviation of the true arrival time. 
```
```suggestion
        the mean squared deviation of the true arrival time. 
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -268,6 +304,19 @@ def get_actual_arrival_time(trip_update: TripUpdate, stop_time_update: StopTimeU
+    Returns
+    -------
+    int | None
+        An int containing the unix timestamp (in seconds) of the actual arrival time.
```
```suggestion
        unix timestamp (in seconds) of the actual arrival time.
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -284,11 +333,16 @@ def experienced_wait_time_delay(trip_stop_time_updates: list[tuple[TripUpdate, S
+    Parameters
+    ----------
+    stop_time_updates : list[tuple[TripUpdate, StopTimeUpdate]]
+        list of corresponding trip updates and stop time updates
```
```suggestion
        pairs of corresponding trip updates and stop time updates
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -284,11 +333,16 @@ def experienced_wait_time_delay(trip_stop_time_updates: list[tuple[TripUpdate, S
+    Returns
+    -------
+    float | None
+        A float containing the average wait time. 
```
```suggestion
       the average wait time
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -479,7 +533,19 @@ def availability_acceptable_vehicle_positions(vehicle_positions: list[VehiclePos
+    timestamp : int
+        An unix timestamp (in seconds) restricting which stop time updates are in the past
+    updates : list[tuple[TripUpdate, StopTimeUpdate]]
+        A list of StopTimeUpdates and corresponding TripUpdates to find the last before the timestamp
```
```suggestion
        pairs of StopTimeUpdates and corresponding TripUpdates to find the last before the timestamp
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -517,11 +592,15 @@ def prediction_reliability(stop_time_updates: list[tuple[TripUpdate, StopTimeUpd
+    Parameters
+    ----------
+    stop_time_updates : list[tuple[TripUpdate, StopTimeUpdate]]
+        list of corresponding trip updates and stop time updates
```
```suggestion
        pairs of corresponding trip updates and stop time updates
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -517,11 +592,15 @@ def prediction_reliability(stop_time_updates: list[tuple[TripUpdate, StopTimeUpd
+    Returns
+    -------
+    float
+        A float containing the prediction reliability.
```
```suggestion
        the prediction reliability.
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -598,12 +677,17 @@ def prediction_inconsistency(actual_arrival_time: int, updates: list[tuple[TripU
+    actual_arrival_time : int
+        An unix timestamp (in seconds), containing the actual arrival time of the vehicle (for that route, trip and stop)
+    updates : list[tuple[TripUpdate, StopTimeUpdate]]
+        list of corresponding trip updates and stop time updates (for that route, trip and stop)
```
```suggestion
        pairs of corresponding trip updates and stop time updates (for that route, trip and stop)
```

1Maxnet1 commented on src/realtime_metrics/main.py:
```diff
@@ -598,12 +677,17 @@ def prediction_inconsistency(actual_arrival_time: int, updates: list[tuple[TripU
+    Returns
+    -------
+    float
+        A float containing the prediction inconsistency.
```
```suggestion
        the prediction inconsistency.
```

1Maxnet1 reviewed (commented):
Looks overall good, thanks for the documentation. I left my comments

WuestFlorian committed:
Apply suggestions from code review

Co-authored-by: Max Buchholz <st172013@stud.uni-stuttgart.de>

WuestFlorian committed:
Adapt links to sections

Adapt links to point to concrete section

1Maxnet1 reviewed (approved):
Thanks, looks good now!

WuestFlorian committed:
Format output logs

Formatted the output logs to be better understandable and added more information, if a metric could not be computed as expected

WuestFlorian commented:
As we discussed, it would be helpful to know, if a result of a metric is due to missing data or the actual result. I added more such logs and formatted the output to be more clear.

WuestFlorian committed:
Change log level

Changed log level to debug in functions called for every trip / stop combination
